.travis.yml

```diff
@@ -184,6 +184,20 @@ install:
 script:
   - travis_wait 45 suite/runsuite_wrapper.pl travis $EXTRA_ARGS
 
+after_script:
+  # Travis logs for OSX builds are truncated because the VM exits early.
+  # Sleep to allow Travis to capture complete log. More context on
+  # DynamoRIO/dynamorio#4223.
+  - test $SLEEP_AFTER_SCRIPT_OR_FAILURE = "yes" && sleep 10
+  - echo "--end--"
+
+after_failure:
+  # Travis logs for OSX builds are truncated because the VM exits early.
+  # Sleep to allow Travis to capture complete log. More context on
+  # DynamoRIO/dynamorio#4223.
```

Comment:
Please update #4223 with the Travis bug link and what you tried and whether it worked (the fcntl, and the sleep).

Reply:
Done.